packages/cli-platform-ios/src/commands/runIOS/index.ts

```diff
@@ -51,25 +51,25 @@ async function runIOS(_: Array<string>, ctx: Config, args: FlagsT) {
   // check if pods need to be installed
   await resolvePods(ctx.root, ctx.dependencies, {forceInstall: args.forcePods});
 
-  const packagerStatus = await isPackagerRunning(port);
-
-  if (
-    typeof packagerStatus === 'object' &&
-    packagerStatus.status === 'running'
-  ) {
-    if (packagerStatus.root === ctx.root) {
-      packager = false;
-      logAlreadyRunningBundler(port);
-    } else {
+  if (packager) {
+    const packagerStatus = await isPackagerRunning(port);
+
+    if (
+      typeof packagerStatus === 'object' &&
+      packagerStatus.status === 'running'
+    ) {
+      if (packagerStatus.root === ctx.root) {
+        packager = false;
+        logAlreadyRunningBundler(port);
+      } else {
+        const result = await handlePortUnavailable(port, ctx.root, packager);
+        [port, packager] = [result.port, result.packager];
+      }
+    } else if (packagerStatus === 'unrecognized') {
```

Comment:
Can see some duplication for both iOS and Android platforms. @szymonrybczak  WDYT about extracting this code to one place? 🤔